### PR TITLE
Human-readable names and update outputs

### LINKED ISSUE #49: Add full names to levels
It would be nice to display full, stylized names like "Getting started" instead of "intro committing". You'll need to add a parameter to the Challenge class, call it something like `self.human_name`. Then, go through the existing challenge and add fun/creative names to each of the levels and then output the human name of the challenge when loading it.

### LINKED ISSUE #140: Match the formatting for `git gud skills` and `git gud levels`
Here is the current output. `git gud levels` should look like `git gud skills`
```
$ git gud skills
Currently on skill: "basics"

Skill 1 - "basics" : 4 levels
    Level  1 : committing
    Level  2 : branching
    Level  3 : merging
    Level  4 : rebasing
Skill 2 - "rampup" : 4 levels
    Level  1 : detaching
    Level  2 : relrefs1
    Level  3 : relrefs2
    Level  4 : reversing
Skill 3 - "extras" : 1 level
    Level  1 : octopus

Load a level with `git gud load`
$ git gud levels
Levels in the current skill "basics" :

1: committing
2: branching
3: merging
4: rebasing
```

### PR DESCRIPTION
**Pull request by sahansk2**
`git gud help load` has been updated to indicate that the shortened identifier should be used to load a level.
`git gud skills` has been updated to be more consistent between identifiers that can be used to load levels, and human-readable names have been added to the output of `git gud skills`, if they are set.
Various initializers have been updated to accommodate human-readable names. 

See `gitgud/skills/rampup` for example initialization of human-readable names. 

Fixes #49 
Fixes #140

### CODE REVIEW

**Comment by benthayer** [comment]
Can you do all this:
```
[ user@laptop ] $ git gud load next
1. The Basics (basics)
└── 1. Introduction to Commits (committing)

Goal: Commit twice using "git gud commit", then test using "git gud test".

<<<<<<< git log --graph --oneline --all
* d4c5b36 (HEAD -> master) 1
=======

 [ user@laptop ] $ git gud levels
1. The Basics (basics)
├── 1. Intro to Commits (committing)
└── 2. Intro to Branching (branching)


 [ user@laptop ] $ git gud levels --skills
1. The Basics (basics)
2. More Advanced (more)


 [ user@laptop ] $ git gud levels --all
1. The Basics (basics)
├── 1. Intro to Commits (committing)
└── 2. Intro to Branching (branching)
2. More Advanced (more)
└── 1. Intro to Branching (branching)


 [ user@laptop ] $ git gud levels --short
1. basics
├── 1. committing
└── 2. branching
```

**Comment by benthayer** [comment]
Just noting that the names are not necessarily correct, you should just focus on the formatting.

You also don't need to do mess with any of the formatting for `git gud load next` other than the stuff that looks like the output of `git gud levels`

File: gitgud/skills/intro/__init__.py [hunk 1]
```diff
@@ -4,8 +4,9 @@
 from gitgud.skills.user_messages import show_tree
 
 class Intro(BasicLevel):
-    def __init__(self):
+    def __init__(self, readable_name=""):
```
Comment (benthayer) [inline gitgud/skills/intro/__init__.py:7]: In some of the new updates, I got rid of the `__init__` method here. You'll see after merging master. We can just set everything via the constructor for `BasicLevel`

File: gitgud/skills/level_builder.py [hunk 2]
```diff
@@ -74,8 +75,8 @@ def test_failed(self):
           
 
 class BasicLevel(Level):
-    def __init__(self, name, skill_package):
-        super().__init__(name)
+    def __init__(self, name, skill_package, readable_name=""):
```
Comment (benthayer) [inline gitgud/skills/level_builder.py:78]: Don't give this a default value. We want everything to have a human readable name

**Comment by benthayer** [comment]
> The more I think about it, the more I think having things closer together is fine in general. For example, when the working tree is clean and there aren't any commands to output, and `git status` is run, the output is packed together:
> 
> ```
> [user@laptop] $ git status
> On branch master
> nothing to commit, working tree clean
> [user@laptop] $
> ```
> 
> For me, in general, having things closer together isn't that inconvenient: more so, it's having to scroll through the terminal in order to recall certain output directions because the output is more spaced out.
> 
> There are limits to this though: I think putting line breaks for (1) separating readable text output [paragraphs] into blocks is fine, and I think putting line breaks for (2) commands if we're explaining them is also fine. (For reference, I'm looking at the output of `git help` for (1), which groups commands based on types of tasks, and I'm looking at the output of `git push` in a fresh directory for (2), where it explains what commands should be used to set upstream and whatnot).
> 
> I also think that putting line breaks for (3) padding git-specific output within delimiters is fine too. See below example output:
> (Note: I use `git gud commit` as an example of a git gud command without any notable git output, and `git gud baz` as an example of a git gud command with notable git output that should be padded).
> 
> ```
> [user@laptop] $ git gud load next
> Skill: The Basics                 (basics)
> Level: Your First Commit          (committing)
> 
> Goal: Commit twice on the master branch. For more foo, type in `git gud bar`.
> After bar'ing, qux three times on bugFix.
> 
> This sentence is only here to illustrate the use of a line break to separate written 
> text from other useful output or unrelated paragraphs.
> [user@laptop] $ git gud commit 
> <<<<<<< Simulating: Create file "2"
> <<<<<<< Simulating: git add 2
> <<<<<<< Simulating: git commit -m "2"
> [INFO]: New Commit: abcdef123 
> [user@laptop] $ git gud baz 
> <<<<<<< Simulating: git log --graph --oneline --all
> 
> * abcdef123 (HEAD -> master) 1
> 
> =======
> [user@laptop] $
> ```
> 
> You could probably tell, but I really _really_ like the git-merge-conflict-style delimiters upside_down_face...but of course, whatever delimiter gets the job done the best is what we should use.
> What do you think, @benthayer ?

From #184.

Given this and what we have above, do you know what to do?

**Comment by sahansk2** [comment]
Yep! I'll comment if I run into any problems. Sorry I haven't tended to this issue, I've been in quite the `pickle` recently.

**Comment by sahansk2** [comment]
Was putting the skill-level tree generation code in `user_messages.py` preferred? I put it there because the entire tree is basically a giant user message that serves no other purpose than to produce output.

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,8 +20,8 @@ jobs:
       - run:
           name: Level synthesis test
           command: |
-            echo -e "y\n" | python3 ./make_level.py testskill testlevel1
-            echo -e "y\n" | python3 ./make_level.py testskill testlevel2
+            echo -e "y\n" | python3 ./make_level.py testskill "Test Skill" testlevel1 "Test Level 1"
+            echo -e "y\n" | python3 ./make_level.py testskill "Test Skill" testlevel2 "Test Level 2"
```
Comment (benthayer) [inline .circleci/config.yml:24]: Let's change the signature to accept something like this
```
make_level.py levelcode "Level Name" skillcode "Skill Name"
```
Skill name is optional/ignored depending on whether `skillcode` already exists

File: gitgud/__main__.py [hunk 4]
```diff
@@ -362,67 +367,56 @@ def handle_solution(self, args):
...
-        else:
+        if args.opt_all or args.opt_skills:
+            if args.opt_all:
+                print("All levels and skills:")
+                print()
+                skills_levels_tree(
+                    [skill for skill in all_skills],
+                    expand_skills=True,
+                    show_human_names=not args.opt_short
+                    )
```
Comment (sahansk2) [inline gitgud/__main__.py:382]: That close parenthesis has one too many tabs. :(

File: make_level.py [hunk 5]
```diff
@@ -96,7 +97,7 @@ def make_skill(skill_name, skill_path):
     print("Registered skill \"{}\" in {}".format(skill_name, filepath))
 
 
-def make_level(skill_name, skill_path, level_name):
+def make_level(skill_name, skill_path, level_name, level_long_name):
```
Comment (benthayer) [inline make_level.py:100]: We'll probably change this definition

File: make_level.py [hunk 6]
```diff
@@ -133,17 +134,19 @@ def create_level_file(level_path, filename):
 
 def get_new_level_name_from_args():
```
Comment (benthayer) [inline make_level.py:135]: This whole method should change with the new definition

**Pull request merged**